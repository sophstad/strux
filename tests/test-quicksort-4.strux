int main() {
	num[9] a = [1.2, 4.4, 3.5, 6.5, 7.5, 2.3, 99.5, 23.9, 37.9];
  a.fshowQuickSort();
	return 0;
}
