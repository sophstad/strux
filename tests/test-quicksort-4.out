=====================================================================================
At this step:
current array:[1.200000 4.400000 3.500000 6.500000 7.500000 2.300000 99.500000 23.900000 37.900000 ]
numbers swapped: 23.900000,6.500000
array after swap: [1.200000 4.400000 3.500000 23.900000 7.500000 2.300000 99.500000 6.500000 37.900000 ]
pivot is 6.500000
numbers swapped: 99.500000,6.500000
array after swap: [1.200000 4.400000 3.500000 23.900000 7.500000 2.300000 6.500000 99.500000 37.900000 ]
numbers swapped: 37.900000,99.500000
array after swap: [1.200000 4.400000 3.500000 23.900000 7.500000 2.300000 6.500000 37.900000 99.500000 ]
=====================================================================================
At this step:
current array:[1.200000 4.400000 3.500000 23.900000 7.500000 2.300000 6.500000 37.900000 99.500000 ]
numbers swapped: 2.300000,3.500000
array after swap: [1.200000 4.400000 2.300000 23.900000 7.500000 3.500000 6.500000 37.900000 99.500000 ]
pivot is 3.500000
numbers swapped: 23.900000,3.500000
array after swap: [1.200000 4.400000 2.300000 3.500000 7.500000 23.900000 6.500000 37.900000 99.500000 ]
numbers swapped: 6.500000,7.500000
array after swap: [1.200000 4.400000 2.300000 3.500000 6.500000 23.900000 7.500000 37.900000 99.500000 ]
=====================================================================================
At this step:
current array:[1.200000 4.400000 2.300000 3.500000 6.500000 23.900000 7.500000 37.900000 99.500000 ]
pivot swapped: 4.400000,3.500000
array after swap: [1.200000 3.500000 2.300000 4.400000 6.500000 23.900000 7.500000 37.900000 99.500000 ]
numbers swapped: 2.300000,3.500000
array after swap: [1.200000 2.300000 3.500000 4.400000 6.500000 23.900000 7.500000 37.900000 99.500000 ]
pivot is 3.500000
=====================================================================================
At this step:
current array:[1.200000 2.300000 3.500000 4.400000 6.500000 23.900000 7.500000 37.900000 99.500000 ]
numbers swapped: 2.300000,1.200000
array after swap: [2.300000 1.200000 3.500000 4.400000 6.500000 23.900000 7.500000 37.900000 99.500000 ]
pivot is 1.200000
=====================================================================================
At this step:
current array:[2.300000 1.200000 3.500000 4.400000 6.500000 23.900000 7.500000 37.900000 99.500000 ]
pivot swapped: 2.300000,1.200000
array after swap: [1.200000 2.300000 3.500000 4.400000 6.500000 23.900000 7.500000 37.900000 99.500000 ]
pivot is 1.200000
=====================================================================================
At this step:
current array:[1.200000 2.300000 3.500000 4.400000 6.500000 23.900000 7.500000 37.900000 99.500000 ]
pivot is 0.000000
=====================================================================================
At this step:
current array:[1.200000 2.300000 3.500000 4.400000 6.500000 23.900000 7.500000 37.900000 99.500000 ]
pivot swapped: 23.900000,7.500000
array after swap: [1.200000 2.300000 3.500000 4.400000 6.500000 7.500000 23.900000 37.900000 99.500000 ]
pivot is 7.500000
